### PR TITLE
OSDOCS-5316: Update microshift docs

### PR DESCRIPTION
**Pull request by eggfoobar**
Version(s):
4.14+

Issue:

https://issues.redhat.com/browse/OSDOCS-5316

Link to docs preview:

https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/understanding-api-support-tiers

QE review:
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: makeBuild.py [hunk 1]
```diff
@@ -99,7 +99,7 @@ def _fix_ids_for_html4(tree):
             #continue
 
         # rest api book is a pain and doesn't convert well
-        if book == "rest_api":
+        if book == "rest_api" or book == "microshift_rest_api":
```
Comment (kalexand-rh) [inline makeBuild.py:102]: @eggfoobar, if you do this, the files can't be rebuilt to be published on the portal. (So you can't do this.)

Comment (eggfoobar) [reply makeBuild.py]: Thanks @kalexand-rh , I removed it and did some more work to try and fix the xrefs, not sure if that's too large. Do you have a way to build a temp portal site like we do with netlify now?

**Comment by ShaunaDiaz** [comment]
@eggfoobar If you are good with the contents, please 
1) remove WIP and add "OSDOCS-5316:" before "Update microshift docs"; 
2) add https://issues.redhat.com/browse/OSDOCS-5316 as the Issue; 
3) add the preview link: https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/understanding-api-support-tiers
4) squash the commits down to 1

**Comment by gangwgr** [comment]
@ShaunaDiaz I didn't see this api in MicroShift. Can you please confirm on this?
https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/autoscaling_apis/scale-autoscaling-v1
```
oc api-resources --loglevel=6|grep -i auto
I0922 19:13:39.787996   91853 loader.go:373] Config loaded from file:  /Users/rahulgangwar/office-work/kubeconfig/kube.txt
I0922 19:13:40.815255   91853 round_trippers.go:553] GET https://dhcp-1-235-237.arm.eng.rdu2.redhat.com:6443/api?timeout=32s 200 OK in 1026 milliseconds
I0922 19:13:41.081642   91853 round_trippers.go:553] GET https://dhcp-1-235-237.arm.eng.rdu2.redhat.com:6443/apis?timeout=32s 200 OK in 258 milliseconds
horizontalpodautoscalers          hpa                 autoscaling/v2                         true         HorizontalPodAutoscaler
```

**Comment by gangwgr** [comment]
Not see this also
io.k8s.api.core.v1.Affinity schema

**Comment by gangwgr** [comment]
I see name in doc Ingress and IngressClass but it Ingresses and IngressClasses
```
https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/networking_apis/networking-apis-index
oc api-resources --loglevel=6|grep -i network
I0922 19:28:37.287449   92244 loader.go:373] Config loaded from file:  /Users/rahulgangwar/office-work/kubeconfig/kube.txt
I0922 19:28:38.363024   92244 round_trippers.go:553] GET https://dhcp-1-235-237.arm.eng.rdu2.redhat.com:6443/api?timeout=32s 200 OK in 1074 milliseconds
I0922 19:28:38.628650   92244 round_trippers.go:553] GET https://dhcp-1-235-237.arm.eng.rdu2.redhat.com:6443/apis?timeout=32s 200 OK in 258 milliseconds
ingressclasses                                        networking.k8s.io/v1                   false        IngressClass
ingresses                         ing                 networking.k8s.io/v1                   true         Ingress
networkpolicies                   netpol              networking.k8s.io/v1                   true         NetworkPolicy
```

**Comment by ShaunaDiaz** [comment]
@eggfoobar Can you PTAL at QE's questions and confirm or address?

**Comment by eggfoobar** [comment]
> @ShaunaDiaz I didn't see this api in MicroShift. Can you please confirm on this? https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/autoscaling_apis/scale-autoscaling-v1
> 
> ```
> oc api-resources --loglevel=6|grep -i auto
> I0922 19:13:39.787996   91853 loader.go:373] Config loaded from file:  /Users/rahulgangwar/office-work/kubeconfig/kube.txt
> I0922 19:13:40.815255   91853 round_trippers.go:553] GET https://dhcp-1-235-237.arm.eng.rdu2.redhat.com:6443/api?timeout=32s 200 OK in 1026 milliseconds
> I0922 19:13:41.081642   91853 round_trippers.go:553] GET https://dhcp-1-235-237.arm.eng.rdu2.redhat.com:6443/apis?timeout=32s 200 OK in 258 milliseconds
> horizontalpodautoscalers          hpa                 autoscaling/v2                         true         HorizontalPodAutoscaler
> ```

Hey @gangwgr so it seems theres a difference between what the api-resources print and what's present in the openapi spec. The openapi spec shows everything that a cluster can support, it seems the api-resources prints preferred versions. If you run `oc get --raw /openapi/v2 | jq . | grep "autoscaling"` you'll see them there. Note, that the same holds true for OCP as well. 

> Not see this also
> io.k8s.api.core.v1.Affinity schema

Affinity is a sub-resource of PodSpec, so this should be fine. 
https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/#io.k8s.api.core.v1.PodSpec

> @ShaunaDiaz Please confirm once about core apis with dev. Not sure about this. https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/ e.g. About https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/#io.k8s.api.core.v1.AWSElasticBlockStoreVolumeSource

Regarding the AWS resources, this is part of the Spec for https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/#io.k8s.api.core.v1.PersistentVolumeSpec So it's technically correct that, that spec supports those items. @pmtk Do we have any limitation on what someone could do with those?

**Comment by pmtk** [comment]
> > @ShaunaDiaz Please confirm once about core apis with dev. Not sure about this. https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/ e.g. About https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/#io.k8s.api.core.v1.AWSElasticBlockStoreVolumeSource
> 
> Regarding the AWS resources, this is part of the Spec for https://64711--docspreview.netlify.app/microshift/latest/microshift_rest_api/objects/#io.k8s.api.core.v1.PersistentVolumeSpec So it's technically correct that, that spec supports those items. @pmtk Do we have any limitation on what someone could do with those?

Nothing that I'm aware of. Looking at our vendored code, it should work.
BTW. I see it's removed in 1.28, so in 4.15 this will disappear

**Review by opayne1 — COMMENTED** [review]
/lgtm

**Pull request merged**